Adding documentation (#716)
bcgov/jag-file-submission: docs/gettingStarted.md

This guide will provide you with all information you need to get started.

> Request a profile

Reviewer: should this be a heading?

```
## Request a profile
```

> Follow the following instructions in [Getting started - Onboarding](onboarding.md)

> Get a oauth token from our Keycloak Server using the following curl

Reviewer: ```suggestion
Get an oauth token from our Keycloak Server using the following curl
```

```bash
curl --location --request POST 'url' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'client_id=[your client id]' \
--data-urlencode 'grant_type=client_credentials' \
--data-urlencode 'client_secret=[your client secret]'
```

> > the secret should be kept safe and not expose to the public

Reviewer: ```suggestion
> the secret should be kept safe and not exposed to the public
```

Start by uploading document(s) to the efiling api

```bash
curl --location --request POST 'http://fla-nginx-proxy-qzaydf-dev.pathfinder.gov.bc.ca/api/submission/documents' \
--header 'X-Transaction-Id: ca09e538-d34e-11ea-87d0-0242ac130003' \
--header 'Content-Type: multipart/form-data' \
--header 'Authorization: Bearer [your bearer token]' \
--form 'files=test.pdf'
```

the response includes a submission id that you will use to generate a redirect url

```json
{
  "submissionId": "5e9492cf-e87e-48b5-ba55-0c198d8edde5",
  "received": 1
}
```

> Then generate a unique url to redirect the users

Reviewer: maybe:
```suggestion
Then generate a unique url to redirect the users to the efiling hub
```

```bash
curl --location --request POST 'http://fla-nginx-proxy-qzaydf-dev.pathfinder.gov.bc.ca/api/submission/5e9492cf-e87e-48b5-ba55-0c198d8edde5/generateUrl' \
--header 'X-Transaction-Id: ca09e538-d34e-11ea-87d0-0242ac130003' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer [your bearer token]' \
--data-raw 'the json payload bellow'
```

payload:

```json
{
  "navigation": {
    "success": {
      "url": "http//somewhere.com"
    },
    "error": {
      "url": "http//somewhere.com"
    },
    "cancel": {
      "url": "http//somewhere.com"
    }
  },
  "clientApplication": {
    "displayName": "your application name"
  },
  "filingPackage": {
    "court": {
      "location": "XXXX",
      "level": "X",
      "courtClass": "X",
      "division": "X"
    },
    "documents": [
      {
        "name": "test.pdf",
        "type": "XXX"
      }
    ]
  }
}
```

The `navigation` object represents a list of possible returns to your application based on the status of the document e-filing

> the `clientAppplication` object represents how your application is labelled in e-filing-hub

Reviewer: ```suggestion
the `clientAppplication` object represents how your application is labelled in efiling hub
```

> the `filingPackage` object represents the court information about the submited package

Reviewer: ```suggestion
the `filingPackage` object represents the court information about the submited package along with the document(s) info
```

> the `documents` array represents the previously uploaded document

Reviewer: ```suggestion
the `documents` array represents the previously uploaded document(s)
```
